pkg/util/test/util.go

```diff
@@ -491,10 +471,70 @@ func validateStorageClusterPods(
 	if _, err := task.DoRetryWithTimeout(t, timeout, interval); err != nil {
 		return err
 	}
+
+	return nil
+}
+
+// Set default Node Affinity rules as Portworx Operator would when deploying StorageCluster
+func addDefaultNodeAffinityRules(cluster *corev1.StorageCluster) *corev1.StorageCluster {
+	cluster.Spec.Placement = &corev1.PlacementSpec{
+		NodeAffinity: &v1.NodeAffinity{
+			RequiredDuringSchedulingIgnoredDuringExecution: &v1.NodeSelector{
+				NodeSelectorTerms: []v1.NodeSelectorTerm{
+					{
+						MatchExpressions: []v1.NodeSelectorRequirement{
+							{
+								Key:      "px/enabled",
+								Operator: v1.NodeSelectorOpNotIn,
+								Values:   []string{"false"},
+							},
+							{
+								Key:      "node-role.kubernetes.io/master",
+								Operator: v1.NodeSelectorOpDoesNotExist,
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	return cluster
+}
+
+func validatePortworxNodes(cluster *corev1.StorageCluster, expectedNodes int) error {
+	conn, err := getSdkConnection(cluster)
+	if err != nil {
+		return nil
+	}
+
+	nodeClient := api.NewOpenStorageNodeClient(conn)
+	nodeEnumerateResp, err := nodeClient.Enumerate(context.Background(), &api.SdkNodeEnumerateRequest{})
+	if err != nil {
+		return err
+	}
+
+	actualNodes := len(nodeEnumerateResp.GetNodeIds())
+	if actualNodes != expectedNodes {
+		return fmt.Errorf("expected nodes: %v. actual nodes: %v", expectedNodes, actualNodes)
+	}
+
+	// TODO: Validate Portworx is started with correct params. Check individual options
+	for _, n := range nodeEnumerateResp.GetNodeIds() {
+		nodeResp, err := nodeClient.Inspect(context.Background(), &api.SdkNodeInspectRequest{NodeId: n})
+		if err != nil {
+			return err
+		}
+		if nodeResp.Node.Status != api.Status_STATUS_OK {
+			return fmt.Errorf("node %s is not online. Current: %v", nodeResp.Node.SchedulerNodeName,
+				nodeResp.Node.Status)
+		}
+
+	}
 	return nil
 }
 
-func expectedPods(cluster *corev1.StorageCluster) (int, error) {
+func getExpectedPodCount(cluster *corev1.StorageCluster) (int, error) {
```

Comment:
I thought we decided to compare the actual node lists instead of count.

Reply:
Done, added more check that we talked about in the morning.